tests/libvmi/factory.go

```diff
@@ -34,30 +34,48 @@ const (
 // NewFedora instantiates a new Fedora based VMI configuration,
 // building its extra properties based on the specified With* options.
 func NewFedora(opts ...Option) *kvirtv1.VirtualMachineInstance {
-	return newFedora(cd.ContainerDiskFedora, opts...)
+	userData := `#!/bin/bash
+	echo "fedora" | passwd fedora --stdin
+	echo `
+
+	return newFedora(cd.ContainerDiskFedora, userData, opts...)
 }
 
 // NewSriovFedora instantiates a new Fedora based VMI configuration,
 // building its extra properties based on the specified With* options, the
 // image used include Guest Agent and some moduled needed by SRIOV.
 func NewSriovFedora(opts ...Option) *kvirtv1.VirtualMachineInstance {
-	return newFedora(cd.ContainerDiskFedoraSRIOVLane, opts...)
+	userData := `#!/bin/bash
+	setenforce 0
+	echo "fedora" | passwd fedora --stdin`
+
+	userData = addStartGuestAgentUserDataSuffix(userData)
+	return newFedora(cd.ContainerDiskFedoraSRIOVLane, userData, opts...)
+}
+
+// Start Guest agent must be that last operation of UserData.
+// This will allows to wait for Agent connected event and to know
+// that applying UserData is finished, and that a login can proceed.
+func addStartGuestAgentUserDataSuffix(userData string) string {
+	userData += `
+	cp /home/fedora/qemu-guest-agent.service /lib/systemd/system/
+	systemctl daemon-reload
+	systemctl start qemu-guest-agent
+	systemctl enable qemu-guest-agent`
```

Comment:
Would you try with `enable --now` it show just work 
[see this related comment on kubevirci PR ](https://github.com/kubevirt/kubevirtci/pull/550#discussion_r582037785)

Reply:
Answered there, lets keep the discussion there please.